src/common/snippets/src/op/reshape.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/itt.hpp"
+
+#include "snippets/op/reshape.hpp"
+#include "snippets/utils.hpp"
+
+
+namespace ov {
+namespace snippets {
+namespace op {
+Reshape::Reshape(const Output<Node>& arg, ov::PartialShape target_shape)
+    : Op({arg}), m_target_shape(target_shape) {
+    constructor_validate_and_infer_types();
+}
+
+void Reshape::validate_and_infer_types() {
+    set_output_type(0, get_input_element_type(0), m_target_shape);
+}
+
+std::shared_ptr<Node> Reshape::clone_with_new_inputs(const OutputVector& new_args) const {
+    INTERNAL_OP_SCOPE(Reshape);
+    check_new_args_count(this, new_args);
+    return std::make_shared<Reshape>(new_args.at(0), get_target_shape());
```

Comment:
Minor. We can explicitly set the target shape here 😃 
```suggestion
    return std::make_shared<Reshape>(new_args.at(0), m_target_shape);
```

Reply:
thanks Alexandra, have done!